Metric/Metric.php

```diff
@@ -198,4 +198,14 @@ private function resolveTagValue(string $valueToResolve, array $resolvers): ?str
                 return $valueToResolve;
         }
     }
+
+    private function correctValue(string $value): string
```

Comment:
Not clear IMHO… It's too easy to "not understand" why/when it's important to call this method… 
**What about fixing the value directly in constructor?** Same `if`, and stored fixed value directly in `$this->paramValue`

Reply:
:thinking: We don't know where comes from the value. It can be a parameter, a property of the event, a function…

Reply:
Not totally satisfied by the design, but it could be the subject of an other PR 👍